source/draw/gpu/GpuCanvas.ooc

```diff
@@ -72,7 +76,9 @@ GpuCanvasYuv420Semiplanar: class extends GpuSurface {
 		this _target y canvas fill()
 		this _target uv canvas fill()
 	}
-	draw: override func ~mesh (image: Image, mesh: GpuMesh) {
+	draw: override func ~mesh (image: GpuImage, mesh: GpuMesh) {
+		if (!image instanceOf?(GpuYuv420Semiplanar))
+			Debug raise("Invalid input format in GpuYuv420SemiplanarCanvas::draw~mesh")
```

Comment:
Is there a clean way to prevent the program from even compiling if anyone tries to call this function with anything but a `GpuYuv420Semiplanar`?